exasol_integration_test_docker_environment/test/test_test_container_reuse.py

```diff
@@ -0,0 +1,169 @@
+from exasol_integration_test_docker_environment.lib.base.docker_base_task import DockerBaseTask
+from exasol_integration_test_docker_environment.lib.data.container_info import ContainerInfo
+from exasol_integration_test_docker_environment.lib.docker.images.clean.clean_images import CleanImagesStartingWith
+from exasol_integration_test_docker_environment.lib.test_environment.prepare_network_for_test_environment import \
+    PrepareDockerNetworkForTestEnvironment
+from exasol_integration_test_docker_environment.lib.test_environment.spawn_test_container import SpawnTestContainer
+
+
+class TestTask(DockerBaseTask):
+    reuse = luigi.BoolParameter()
+    attempt = luigi.IntParameter()
+
+    def run_task(self):
+        docker_network_task_1 = PrepareDockerNetworkForTestEnvironment(
+            environment_name="test_environment_TestContainerReuseTest",
+            network_name="docker_network_TestContainerReuseTest",
+            test_container_name="test_container_TestContainerReuseTest",
+            db_container_name="db_container_TestContainerReuseTest",
+            reuse=self.reuse,
+            no_cleanup_after_success=True,
+            no_cleanup_after_failure=False,
+            attempt=self.attempt
+        )
+        self.docker_network_future_1 = yield from self.run_dependencies(docker_network_task_1)
+
+        test_container_task_1 = \
+            SpawnTestContainer(
+                environment_name="test_environment_TestContainerReuseTest",
+                test_container_name="test_container_TestContainerReuseTest",
+                network_info=self.docker_network_future_1.get_output(),
+                ip_address_index_in_subnet=2,
+                attempt=self.attempt,
+                reuse_test_container=self.reuse,
+                no_test_container_cleanup_after_success=True,
+                no_test_container_cleanup_after_failure=False
+            )
+        test_container_future_1 = yield from self.run_dependencies(test_container_task_1)
+        container_info = test_container_future_1.get_output()  # type: ContainerInfo
+        container = self._client.containers.get(container_info.container_name)
+
+        self.return_object({"container_id": container.image.id, "image_id": container.image.id})
+
+
+class TestContainerReuseTest(unittest.TestCase):
+
+    def set_job_id(self, task_cls):
+        strftime = datetime.now().strftime('%Y_%m_%d_%H_%M_%S')
+        job_id = f"{strftime}_{task_cls.__name__}"
+        config = luigi.configuration.get_config()
+        config.set('job_config', 'job_id', job_id)
+        return job_id
+
+    def clean(self):
+        jobid = self.set_job_id(CleanImagesStartingWith)
+        task = CleanImagesStartingWith(starts_with_pattern="", jobid=jobid)
+        luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+        if task._get_tmp_path_for_job().exists():
+            shutil.rmtree(str(task._get_tmp_path_for_job()))
```

Comment:
Won't that be removed together with self.temp_directory?

Reply:
Got removed anyway and replaced by task.cleanup() which does more or less the same + other things. I wanted to have it to remove temporary files, such that there is no chance of influence, but to be honest, it is probably unnecessary